python/tvm/runtime/module.py

```diff
@@ -230,8 +288,7 @@ def evaluator(*args):
                 blob = feval(*args)
                 fmt = "@" + ("d" * repeat)
                 results = struct.unpack(fmt, blob)
-                mean = sum(results) / float(repeat)
-                return ProfileResult(mean=mean, results=results)
+                return BenchmarkResult(results)
```

Comment:
Does time_evaluator itself have a unit test that needs to be updated? If not, would it make sense to add a quick one?

Reply:
time_evaluator was not changed, so it makes no sense to create/modify a unit test for it.

Reply:
Um, unless I'm reading this diff incorrectly (sorry if I'm getting confused by github diffs!) this is a change to time_evaluator. Am I confused?

Reply:
Oh yes, this is the python binding for time_evaluator. Output is essentially the same, so there is no need to modify the tests.

Reply:
Well, OK, but the existing tests for time_evaluator seem kind of weak. Here we're adding a new class (BenchmarkResult) which calculates a number of new statistics, but there's no tests for it ... meaning ... a regression in this behavior (either a change to time_evaluator or BenchmarkResult) would not be caught by CI. 

At minimum I'd recommend a unit test for BenchmarkResult() itself, which is easy to write. Not trying to be too nitpicky but I do feel like we should be constantly improving our test coverage and adding a new class without corresponding tests tends to be a red flag for me :-)

WDYT?

Reply:
To clarify `BenchmarkResult` is just a renaming + adding a constructor to `ProfileResult` (which already existed). It is tested in an ad hoc manner in a variety of places. I've added some tests for it constructor.